src/main/java/com/uid2/shared/vertx/RequestCapturingHandler.java

```diff
@@ -75,8 +78,11 @@ private void capture(RoutingContext context, long timestamp, String remoteClient
 
         String path = null;
         try {
-            path = context.currentRoute().getPath();
-        } catch (NullPointerException ex) {
+            // If the current route is a known path, extract the full path from the request URI
+            if (context.currentRoute().getPath() != null) {
```

Comment:
What is the reason for this check?

Reply:
If the router doesn't recognise the path, it's null. There's some logic below that sets the metric's path tag to "unknown" if the path is null. So this helps preserve that existing logic